File: google_foo_bar/numbers_station_coded_messages.py

def solution(arr, num):
    for i in range(len(arr)):
        sum = arr[i]
        if(sum == num):
            return [i, i]
        for j in range(i+1, len(arr)):
            sum += arr[j]
            if(sum == num):
                return [i, j]

    return [-1, -1]



print(solution([4,3,5,7,8], 12))
print(solution([1,2,3,4], 15))
print(solution([4,3,10,2,8], 12))
print(solution([6,2,3], 6))

File: google_foo_bar/numbers_station_coded_messages.js

function solution(arr, num) {
    for(let i=0; i<arr.length-1; i++) {
        let sum = arr[i]
        for(let j=i+1; j<arr.length; j++) {
            sum += arr[j]
            if(sum == num) {
                return [i,j]
            }
        }
    }
    return [-1, -1]
}


console.log(solution([4,3,5,7,8], 12))
console.log(solution([1,2,3,4], 15))
console.log(solution([4,3,10,2,8], 12))